openupgrade_scripts/scripts/base_address_extended/16.0.1.1/pre-migration.py

```diff
@@ -49,7 +49,7 @@ def migrate(env, _version):
         FROM (
             SELECT
                 id,
-                regexp_match(street, '^(.*?)(?:\\s([0-9][0-9\\S]*))?(?: - (.+))?$') AS rgx
```

Comment:
I think the problem is with `\S`, see:

In v13: https://www.postgresql.org/docs/13/functions-matching.html
```
\S 	[^[:space:]]
Within bracket expressions, \d, \s, and \w lose their outer brackets, and \D, \S, and \W are illegal. (So, for example, [a-c\d] is equivalent to [a-c[:digit:]]. Also, [a-c\D], which is equivalent to [a-c^[:digit:]], is illegal.)
```

In v15: https://www.postgresql.org/docs/15/functions-matching.html
```
\S 	matches any non-whitespace character, like [^[:space:]]
The class-shorthand escapes also work within bracket expressions, although the definitions shown above are not quite syntactically valid in that context. For example, [a-c\d] is equivalent to [a-c[:digit:]].
```

Reply:
Following @MiquelRForgeFlow comments, we can use `regexp_match(street, '^(.*?)(?:\\s([0-9][0-9^\\s]*))?(?: - (.+))?$') AS rgx`

When tested, both queries where not giving the same result (tested on 13)

Reply:
and was the same result expected on postgres 15